packages/frontend/src/app/submissions/page.tsx

```diff
@@ -0,0 +1,133 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+import Image from 'next/image'
+import dayjs from 'dayjs'
+import { PageHeading } from '@/common/components/PageHeading'
+import { Text4xlBold } from '@/common/components/Typography'
+import { Submission } from '@/common/interfaces'
+import { PageContainer } from '@/common/components/PageContainer'
+import { Table, TableBody, TableHeader } from '@/common/components/Table'
+import { Filter, FilterButton } from '@/common/components/Filter'
+import { betterFetch } from '@/api'
+
+export default function Page() {
+  const [submissions, setSubmissions] = useState<Submission[]>([])
+  const [filterValues, setFilterValues] = useState<string[]>([])
+
+  function toggleFilterValue(filter: string): void {
+    if (filterValues.includes(filter)) {
+      const filterIndex = filterValues.indexOf(filter)
+
+      filterValues.splice(filterIndex, 1)
+      setFilterValues([...filterValues])
+    } else {
+      setFilterValues([...filterValues, filter])
+    }
+  }
+
+  function isActive(filter: string): boolean {
+    return filterValues.includes(filter)
+  }
+
+  useEffect(() => {
+    try {
+      betterFetch('GET', 'http://localhost:3000/api/submissions').then(
+        (submissions) => {
+          setSubmissions(submissions)
+        },
+      )
+    } catch (error) {
+      setSubmissions([])
+    }
```

Comment:
The `betterFetch` function is not awaited, so I'm not sure about the `catch` here. If I understand it correctly the error from `betterFetch` is not handled at all.

Reply:
You are right. I'll fit it.